chia/wallet/wallet_state_manager.py

```diff
@@ -1047,14 +1043,16 @@ async def add_coin_states(
                         wallet_type = local_record.wallet_type
                     elif coin_state.created_height is not None:
                         wallet_id, wallet_type = await self.determine_coin_type(peer, coin_state, fork_height)
-                        potential_dl = self.get_dl_wallet()
-                        if potential_dl is not None:
+                        try:
+                            dl_wallet = self.get_dl_wallet()
                             if (
-                                await potential_dl.get_singleton_record(coin_name) is not None
+                                await dl_wallet.get_singleton_record(coin_name) is not None
                                 or coin_state.coin.puzzle_hash == MIRROR_PUZZLE_HASH
                             ):
-                                wallet_id = potential_dl.id()
-                                wallet_type = potential_dl.type()
+                                wallet_id = dl_wallet.id()
+                                wallet_type = dl_wallet.type()
```

Comment:
This probably belongs in the `else:` or such so you don't start consuming `ValueError` from it.  `try:` blocks should be minimal.

Reply:
See 471aedc9778bff68ee48e11ac23fb000742d4e95.